File: civitasJava/src/civitas/CasillaJuez.java
/**
 * @file CasillaImpuesto.java
 * @author Yesenia González Dávila
 * @author Esther García Gallego
 * @brief Casilla de tipo juez. Subclase de Casilla.
 * @note Grupo B.3
*/

package civitas;
import java.util.ArrayList;

public class CasillaJuez extends Casilla {
    
    /** Constructor juez. */
    CasillaJuez (String nombre, int numCasillaCarcel) {
        super (nombre);
        carcel = numCasillaCarcel;
        tipoCasilla = "JUEZ";
    }
    
    /** Recibe al jugador en la casilla aplicando el proceso ç 
     *  correspondiente al juez. */
    @Override
    void recibeJugador (int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto (actual, todos)) {
            informe (actual, todos);
            todos.get(actual).encarcelar(carcel);
        }
    }
}

File: civitasRuby/lib/civitas/casilla_juez.rb
#encoding:utf-8

=begin
Authors: Esther García Gallego
         Yesenia González Dávila
         Grupo B3
=end

module Civitas
  class CasillaJuez < Casilla
    
    def initialize(numCasillaCarcel, nombre)
      super(nombre)
      @tituloPropiedad = nil
      @importe = -1
      @Carcel = numCasillaCarcel
      @mazo = nil
      @sorpresa = nil
    end
    
    # Override 
    def recibeJugador(actual, todos)
      if(jugadorCorrecto(actual, todos))
        informe(actual, todos)
        todos[actual].encarcelar(@Carcel)
      end
    end
    
  end
end
